internal/cluster/distribution/eks/eksworkflow/workflow_update_node_pool.go

```diff
@@ -130,6 +132,7 @@ func (w UpdateNodePoolWorkflow) Execute(ctx workflow.Context, input UpdateNodePo
 
 	effectiveImage := input.NodeImage
 	effectiveVolumeSize := input.NodeVolumeSize
+	var effectiveSecurityGroups []string
 	if effectiveImage == "" ||
 		effectiveVolumeSize == 0 { // Note: needing CF stack for original information for version.
```

Comment:
**Required**: the `effectiveSecurityGroups` **MUST** take the value of the passed in security group list by default and only retrieve the original value when no new value is specified.

```
effectiveSecurityGroups := input.SecurityGroups
if effectiveImage == "" ||
    effectiveVolumeSize == 0 ||
    effectiveSecurityGroups == nil { // Note: needing CF stack for original information for version.
```
(cannot put suggestion on this block properly)

Reply:
fixed